### DISCUSSION TITLE
Windows Ink Compatibility Issue

### DISCUSSION THREAD

**Discussion by PixelQubed** [ClassicOldSong/Apollo#172] [General]
Hello, I am having a relatively recent issue. 

Windows Ink will confuse the mouse passthrough with the pen passthrough when using an iPad.
I am using the Moonlight ZWM client on iOS while using an iPad Pro 11 [M4].
When trying to click a drop-down or something with the pen, it auto snaps back to where the mouse was last located on the screen.
This can be very frustrating when using a program like Blender. 

The behaviour should be when the pen is used and moved away from the screen; it should just not default right back to the Windows cursor unless the cursor has specifically been called upon by the client, i.e. the use of the trackpad.

I do hope this issue gets fixed soon and the haptic buttons and the rotation functionality are passed through in the future.

If you would like me to demo the issue via a recording, let me know. I am running the latest version of Apollo.

**Comment by ClassicOldSong** [comment]
I think this is a client side issue. I can click the drop down just fine using a stylus with an Android tablet.

Advanced pen features require modifications from both client side and host side so it will not likely to be implemented soon.

**Comment by PixelQubed** [comment]
> I think this is a client side issue. I can click the drop down just fine using a stylus with an Android tablet.
> 
> Advanced pen features require modifications from both client side and host side so it will not likely to be implemented soon.

Damn... Is proper iPad/Apple Pencil Pro support expected in the future, though?

**Comment by ClassicOldSong** [comment]
I just got my hands on a Mac mini M4 and I have an iPad M1. I remember the Apple Pencil has upgraded during the time so I might not be able to test the new features on the new models.

But I get the M4 mac for another project, and I'm currently maintaining Apollo/Artemis for free, so no guarantee there will be any updates for the iOS client any time soon.

**Comment by PixelQubed** [comment]
> I just got my hands on a Mac mini M4 and I have an iPad M1. I remember the Apple Pencil has upgraded during the time so I might not be able to test the new features on the new models.
> 
> But I get the M4 mac for another project, and I'm currently maintaining Apollo/Artemis for free, so no guarantee there will be any updates for the iOS client any time soon.

If you need me to test Artemis when it comes to iOS, just let me know. I would be happy to do so. Also I appreciate that you are doing this all for free and I completely understand official support may not come around soon. I was just asking in general about it being planed for when ever you have the time to do so. 

Honestly I don't understand why Sunshine didn't implement any support since there are artists like me looking for a good streaming program that is good on performance. I have personally tried Duet, AstroPad, Easy Canvas, Parsec and Splashtop.

Splashtop had the best pen support but it was horrible performance wise and the others where just not too good. 

Anyways I can deal with the current support for now since its better than nothing.

**Comment by rp1231** [comment]
I'd be willing to test ios as well.
Apple pencil and touch gestures support would be great!
For example being able to use multi touch gestures in photoshop to rotate and move the canvas....
